tree/BinarySearchTree.js

```diff
@@ -0,0 +1,57 @@
+
+const Node = require('./Node');
+
+class BST {
+  constructor(root = null){
+    this.root = root;
+  }
+
+  add(value){
+    const node = this.root;
+    if(node === null) {
```

Comment:
to make your code more DRY, you can refactor these two lines to:
if(this.root === null)